translate/compiler.go

```diff
@@ -303,13 +303,36 @@ func compileIntegerLiteral(e *ast.IntegerLiteral, bytecode *Bytecode) error {
 	return nil
 }
 
-// TODO: implement me w/ test cases :-)
-func compileStringLiteral(e *ast.StringLiteral, bytecode *Bytecode) error {
+func compileStringLiteral(e *ast.StringLiteral, bytecode *Bytecode, memDefiner MemDefiner) error {
+	operand, err := encoding.EncodeOperand(e.Value)
+	if err != nil {
+		return err
+	}
+
+	for len(operand) >= 8 {
+		memEntry := memDefiner.Define(e.Value)
```

Comment:
In `Define(id string)` , `id` means a key like variable name, not a value. When `int a` is saved in memory, compiling string `"a"` overwrites the memory table. Memory table just saves the location and size of value.
Also, The string literal doesn't have any meaning alone. The statement like assign statement has some meaning.

So, i think it should be
```go
size, err := encoding.EncodeOperand(len(operand))
if err != nil {
        return err
}

offset, err := encoding.EncodeOperand(memDefiner.MemoryCounter)
if err != nil {
        return err
}
```

If so, `Define()` is not used in here. Then it seems better that `memDefiner` is changed to `memEntryGetter` or `memTracer`.

Reply:
Now memEntryGetter and memTracer both don't produce MemCounter because of abstraction.
How about use memTracer global var?